src/mrpro/utils/_databuffermixin.py

```diff
@@ -0,0 +1,104 @@
+from collections import OrderedDict
+
+import torch
+
+from mrpro.data._Data import Data
+
+
+class DataBufferMixin(torch.nn.Module):
+    """A Mixin that allow to set a mrpro.data.Data as a Buffer.
+
+    This implements the same logic for mrpro.Data as torch.nn.Module
+    uses for Buffers, but stores them in a separate dictionary called
+    _data
+
+    The main use is to allow Data objects to be automatically moved to a
```

Comment:
...to allow Data.data to be automatically moved... (we are not moving Data.header at the moment)